modules/xp/rank.ts

```diff
@@ -106,6 +117,8 @@ export default async function getUserRank(
 			},
 		],
 
-		files: [{ attachment: canvas.toBuffer("image/png"), name: "progress.png" }],
+		files: constants.canvasEnabled
+			? [{ attachment: (await makeCanvas())!.toBuffer("image/png"), name: "progress.png" }]
```

Comment:
I would prefer to not use `!.` and instead move this whole conditional inside `makeCanvas` (so it always returns an array)